LanguageFeatures/Augmentation-libraries/applying_augmentation_library_A04_t01.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - The main library and its augmentations do not all have the same language
+///   version. There is only one user-visible library at the end, and it should
+///   have a consistent version across its entire surface area.
+///
+/// @description Check that it is a compile-time error if an augmentation
+/// library has version lower than main.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
```

Comment:
Is it even possible to use different language versions, are there several versions where augmentation is supported?

If we do have enough usable language versions then I think it would be safer to make the language version explicit in all participating libraries.

Reply:
Similarly for applying_augmentation_library_A04.